src/code/fault_n64.c

```diff
@@ -0,0 +1,845 @@
+
+    for (i = 0; i < h; i++) {
+        j = 0;
+        for (j = 0; j < w; j++) {
+            *fbPtr = color;
+            fbPtr++;
+        }
+        fbPtr += gFaultMgr.fbWidth - w;
+    }
+    osWritebackDCacheAll();
+}
+
+void Fault_DrawRecBlack(s32 x, s32 y, s32 w, s32 h) {
+    Fault_DrawRec(x, y, w, h, GPACK_RGBA5551(0, 0, 0, 1));
+}
+
+void Fault_DrawCharImpl(s32 x, s32 y, char c) {
+    u32* dataPtr;
+    s32 shift = c % 4;
+    s32 i;
+    s32 j;
+    u16* fbPtr;
+    u32 data;
+
+    dataPtr = (u32*)sFaultDrawerFont + ((c / 8 * 0x10) + ((c & 4) >> 2));
+    fbPtr = gFaultMgr.fb + (gFaultMgr.fbWidth * y) + x;
+
+    for (i = 0; i < 8; i++) {
+        u32 mask;
+
+        mask = (1 << 28) << shift;
+        data = *dataPtr;
+        for (j = 0; j < 8; j++) {
+            if (mask & data) {
+                *fbPtr = sFaultFontColor;
+            }
+            fbPtr++;
+            mask >>= 4;
+        }
+        dataPtr += 2;
+        fbPtr += gFaultMgr.fbWidth - 8;
+    }
+    osWritebackDCacheAll();
+}
+
+void Fault_DrawChar(s32 x, s32 y, u8 c) {
+    Fault_DrawCharImpl(x, y, c);
+}
+
+void Fault_DrawCornerRec(u16 color) {
+    Fault_DrawRec(22, 16, 8, 1, color);
+}
+
+void Fault_DrawCornerRecRed(void) {
+    Fault_DrawCornerRec(GPACK_RGBA5551(255, 0, 0, 1));
+}
+
+void Fault_DrawCornerRecYellow(void) {
+    Fault_DrawCornerRec(GPACK_RGBA5551(255, 255, 0, 1));
+}
+
+void func_800AE05C(void) {
+}
+
+void* Fault_PrintCallbackDraw(void* arg, const char* str, size_t len) {
+    FaultCursorCoords* coords = arg;
+
+    for (; len != 0; len--, str++) {
+        if (*str == '\n') {
+            coords->x = 320;
+        } else {
+            Fault_DrawChar(coords->x, coords->y, *str);
+            coords->x += 6;
+        }
+        if (coords->x >= 277) {
+            coords->x = 22;
+            coords->y += 8;
+            if (coords->y >= 209) {
+                Fault_WaitForInputImpl();
+                Fault_DrawRecBlack(22, 16, 276, 208);
+                coords->y = 16;
+            }
+        }
+    }
+    return coords;
+}
+
+void Fault_DrawText(s32 x, s32 y, const char* fmt, ...) {
+    va_list args;
+    FaultCursorCoords coords;
+
+    va_start(args, fmt);
+
+    coords.x = x - 8;
+    coords.y = y;
+    _Printf(Fault_PrintCallbackDraw, &coords, fmt, args);
+
+    va_end(args);
+}
+
+void Fault_SetCursor(s32 x, s32 y) {
+    sFaultCursorPos.x = x;
+    sFaultCursorPos.y = y;
+}
+
+void func_800AE1F8(void) {
+    Fault_DrawRecBlack(22, 16, 276, 208);
+    sFaultCursorPos.x = 22;
+    sFaultCursorPos.y = 16;
+}
+
+s32 Fault_Printf(const char* fmt, ...) {
+    s32 ret;
+    va_list args;
+
+    va_start(args, fmt);
+    ret = _Printf(Fault_PrintCallbackDraw, &sFaultCursorPos, fmt, args);
+    va_end(args);
+
+    return ret;
+}
+
+void Fault_PrintFReg(s32 x, s32 y, s32 idx, f32* value) {
+    u32 raw = *(u32*)value;
+    s32 exp = ((raw & 0x7F800000) >> 23) - 127;
+
+    if (((exp >= -0x7E) && (exp < 0x80)) || (raw == 0)) {
```

Comment:
```suggestion
    if (((exp > -127) && (exp <= 127)) || (raw == 0)) {
```
If the subtraction is decimal, feel like this check should also be decimal. Also more nitpicky adjusting the checks to match the subtraction value.